unittests/Dialect/ESI/runtime/ESIRuntimeTest.cpp

```diff
@@ -0,0 +1,449 @@
+  std::vector<uint8_t> bitsValue = {0xAB};
+  std::any validBits = std::any(bitsValue);
+  EXPECT_NO_THROW(bitsType.isValid(validBits));
+
+  // Test invalid size
+  std::vector<uint8_t> wrongSize = {0xAB, 0xCD};
+  std::any invalidBits = std::any(wrongSize);
+  EXPECT_THROW(bitsType.ensureValid(invalidBits), std::runtime_error);
+
+  // Test serialization
+  MessageData serialized = bitsType.serialize(validBits);
+  EXPECT_EQ(serialized.getSize(), 1UL);
+  EXPECT_EQ(serialized.getData()[0], 0xAB);
+
+  // Test deserialization
+  auto [deserialized, remaining] = bitsType.deserialize(serialized);
+  auto deserializedBits = std::any_cast<std::vector<uint8_t>>(deserialized);
+  EXPECT_EQ(deserializedBits.size(), 1UL);
+  EXPECT_EQ(deserializedBits[0], 0xAB);
+  EXPECT_EQ(remaining.size(), 0UL);
+}
+
+// Test UIntType serialization and deserialization
+TEST(ESITypesTest, UIntTypeSerialization) {
+  UIntType uintType("uint16", 16);
+
+  // Test valid uint value
+  uint64_t uintValue = 0x1234;
+  std::any validUInt = std::any(uintValue);
+  EXPECT_NO_THROW(uintType.isValid(validUInt));
+
+  // Test out of range value
+  uint64_t outOfRange = 0x10000; // Too big for 16-bit
+  std::any invalidUInt = std::any(outOfRange);
+  EXPECT_THROW(uintType.ensureValid(invalidUInt), std::runtime_error);
+
+  // Test serialization (little-endian)
+  MessageData serialized = uintType.serialize(validUInt);
+  EXPECT_EQ(serialized.getSize(), 2UL);
+  EXPECT_EQ(serialized.getData()[0], 0x34); // Little-endian
+  EXPECT_EQ(serialized.getData()[1], 0x12);
+
+  // Test deserialization
+  auto [deserialized, remaining] = uintType.deserialize(serialized);
+  auto deserializedUInt = std::any_cast<uint64_t>(deserialized);
+  EXPECT_EQ(deserializedUInt, 0x1234UL);
+  EXPECT_EQ(remaining.size(), 0UL);
+}
+
+// Test SIntType serialization and deserialization
+TEST(ESITypesTest, SIntTypeSerialization) {
+  SIntType sintType("sint16", 16);
+
+  // Test valid positive sint value
+  int64_t positiveValue = 0x1234;
+  std::any validSInt = std::any(positiveValue);
+  EXPECT_NO_THROW(sintType.isValid(validSInt));
+
+  // Test valid negative sint value
+  int64_t negativeValue = -1000;
+  std::any validNegSInt = std::any(negativeValue);
+  EXPECT_NO_THROW(sintType.isValid(validNegSInt));
+
+  // Test serialization of positive value
+  MessageData serialized = sintType.serialize(validSInt);
+  EXPECT_EQ(serialized.getSize(), 2UL);
+  EXPECT_EQ(serialized.getData()[0], 0x34); // Little-endian
+  EXPECT_EQ(serialized.getData()[1], 0x12);
+
+  // Test deserialization
+  auto [deserialized, remaining] = sintType.deserialize(serialized);
+  auto deserializedSInt = std::any_cast<int64_t>(deserialized);
+  EXPECT_EQ(deserializedSInt, 0x1234);
+  EXPECT_EQ(remaining.size(), 0UL);
+
+  // Test negative value serialization/deserialization
+  MessageData negSerialized = sintType.serialize(validNegSInt);
+  auto [negDeserialized, negRemaining] = sintType.deserialize(negSerialized);

```

Comment:
Please add a check of the serialized bytes like are in the unsigned cases above.